packages/spectral/src/serverTypes/convertIntegration.ts

```diff
@@ -365,7 +365,9 @@ const convertComponentReference = (
                 type: "value",
                 value: JSON.stringify(v),
               }))
-            : value.value;
+            : manifestEntryInput.collection === "valuelist" && Array.isArray(value.value)
```

Comment:
Do we also need to support `keyvaluelist`? I can't remember what the expected CNI input value is for those input types.

Reply:
Yep! The preceding lines handle `keyvaluelist` inputs.

Reply:
Well call me bamboozled how we missed supporting this originally. Nice find!